### PR TITLE
Integration test federation node setup and leader test.

### PR DESCRIPTION
**Pull request by ferdeen**
Hi, this is my initial commit.

Wanted to get this in then carry on with the leader test.  

In the leader test I've set `node.EnableFastMining();` but still see the _MISS means that miner didn't produce a block at the timestamp he was supposed to_ .

image

Wanted to get this PR is so other devs could start to work on any integration type tests.

### CODE REVIEW

File: src/Stratis.FederatedPeg.Features.FederationGateway/FederationGatewayFeature.cs [hunk 1]
```diff
@@ -166,9 +166,12 @@ public override async Task InitializeAsync()
             this.maturedBlockRequester.Start();
 
             // Connect the node to the other federation members.
-            foreach (var federationMemberIp in this.federationGatewaySettings.FederationNodeIpEndPoints)
+            if (this.federationGatewaySettings.FederationNodeIpEndPoints != null)
```
Comment (ferdeen) [inline src/Stratis.FederatedPeg.Features.FederationGateway/FederationGatewayFeature.cs]: Add this after starting a node in integration, as `FederationNodeIpEndPoints` can be null at startup.

File: src/Stratis.FederatedPeg.Features.FederationGateway/LeaderProvider.cs [hunk 2]
```diff
@@ -23,24 +23,24 @@ public class LeaderProvider : ILeaderProvider
...
-
         public LeaderProvider(IFederationGatewaySettings federationGatewaySettings)
         {
             this.orderedFederationPublicKeys = federationGatewaySettings.FederationPublicKeys.
                 Select(k => k.ToString()).
                 OrderBy(j => j).
                 ToList().
                 AsReadOnly();
+
+            this.CurrentLeader = new PubKey(this.orderedFederationPublicKeys.First());
```
Comment (ferdeen) [inline src/Stratis.FederatedPeg.Features.FederationGateway/LeaderProvider.cs]: Also here `this.CurrentLeader` is null at startup.  Now it is assigned to a fed member.

**Comment by monsieurleberre** [comment]
2 things:
- it looks like no block is being produced, but the output saying "MISS means ..." Is shown regardless. I think it is an explanatory note rather than a warning.
- leader is in charge of cross chain transfers, not block production, these are slightly different things... It is possible maybe to imagine a federation of 11 gateway nodes with only 5 block producers, although the code is currently using the same keys for signing blocks and withdrawal from multisig :)

**Comment by ferdeen** [comment]
Are okay - So I'll setup the leader test with one mining node (block producer), and one fed gateway node.  When the block's produced the test will assert the leader did indeed change.

**Pull request closed without merging**